packages/lib/ren-tx/src/configs/genericBurn.ts

```diff
@@ -0,0 +1,201 @@
+
+const burnChainMap: BurnMachineContext["fromChainMap"] = {
+    ethereum: (context): MintChain<any> => {
+        return Ethereum(context.providers.ethereum, RenNetwork.Testnet).Account(
+            {
+                address: context.tx.userAddress,
+                value: context.tx.suggestedAmount,
+            }
+        ) as MintChain<any>;
+    },
+};
+
+const releaseChainMap: BurnMachineContext["toChainMap"] = {
+    bitcoin: (context): LockChain => {
+        return Bitcoin().Address(context.tx.destAddress) as any;
+    },
+};
+*/
+
+const burnAndRelease = async (context: BurnMachineContext) => {
+    const transaction = Object.keys(context.tx.transactions)[0];
+    return await context.sdk.burnAndRelease({
+        asset: context.tx.sourceAsset.toUpperCase(),
+        to: context.toChainMap[context.tx.destChain](context),
+        from: context.fromChainMap[context.tx.sourceChain](context),
+        ...(transaction ? { transaction } : {}),
+    });
+};
+
+// Format a transaction and prompt the user to sign
+const txCreator = async (
+    context: BurnMachineContext,
+    // eslint-disable-next-line @typescript-eslint/require-await
+): Promise<GatewaySession> => {
+    const { targetAmount, sourceAsset, sourceChain, destChain } = context.tx;
+    const to = context.toChainMap[destChain](context);
+    const from = context.fromChainMap[sourceChain](context);
+    const decimals = await to.assetDecimals(sourceAsset.toUpperCase());
+
+    let suggestedAmount = new BigNumber(Number(targetAmount) * 10 ** decimals);
+    try {
+        const fees = await context.sdk.getFees({
+            asset: sourceAsset.toUpperCase(),
+            from,
+            to,
+        });
+
+        suggestedAmount = suggestedAmount
+            .plus(fees.release || 0)
+            .plus(suggestedAmount.multipliedBy(fees.burn * 0.001));
+    } catch (error) {
+        // Ignore error
+    }
+
+    const newTx: GatewaySession = {
+        ...context.tx,
+        suggestedAmount: suggestedAmount.decimalPlaces(0).toFixed(),
+    };
+    context.tx = newTx;
+
+    return {
+        ...newTx,
+        transactions: {},
+    };
+};
+
+const spawnBurnTransaction = assign<BurnMachineContext, BurnMachineEvent>({
+    burnListenerRef: (c: BurnMachineContext, _e: any) => {
+        const actorName = `${c.tx.id}BurnListener`;
+        if (c.burnListenerRef) {
+            console.warn("listener already exists");
+            return c.burnListenerRef;
+        }
+        const cb = burnTransactionListener(c);
+        return spawn(cb, actorName);
+    },
+});
+
+const burnTransactionListener = (context: BurnMachineContext) => (
+    callback: Sender<BurnMachineEvent>,
+    receive: Receiver<any>,
+) => {
+    const cleaners: Array<() => void> = [];
+    let burning = false;
+    burnAndRelease(context)
+        .then((burn) => {
+            if (context.autoSubmit) {
+                setTimeout(() => callback("SUBMIT"), 500);
```

Comment:
What is the `setTimeout` for?

Reply:
We need to ensure that the burn actor is ready to receive events before sending the event; 500ms is definitely excessive, setImmediate might even work; but didn't want to play around with it too much at this point